assets/javascripts/kitten/components/form/field.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component, Fragment } from 'react'
+import PropTypes from 'prop-types'
+import { FieldLabel } from 'kitten/components/form/field/label'
+import { FieldInput } from 'kitten/components/form/field/input'
+import { FieldError } from 'kitten/components/form/field/error'
+import { FieldRadioButtonSet } from 'kitten/components/form/field/radio-button-set'
+import { FieldSelect } from 'kitten/components/form/field/select'
+
+export class Field extends Component {
+  static Label = FieldLabel
+  static Input = FieldInput
+  static RadioButtonSet = FieldRadioButtonSet
+  static Select = FieldSelect
+  static Error = FieldError
+
+  static propTypes = {
+    error: PropTypes.boolean,
+  }
+
+  static defaultProps = {
+    error: false,
+  }
+
+  elementsOf = (type, props) => {
+    const { children } = this.props
+
+    return React.Children.toArray(children).map(child => {
+      if (child.type === type) {
+        if (!props) return child
+
+        return React.cloneElement(child, props)
+      }
+    })
+  }
+
+  render() {
+    const { error } = this.props
+
+    return (
+      <Fragment>
```

Comment:
Pourquoi doit-on faire tout ce traitement avec `this.elementsOf` ? et pas seulement afficher les `children` ?
Du coup quel est l'intérêt du `field.js` ? il n'y a aucun style dessus ? Codé comme ceci j'ai l'impression qu'on peut s'en passer.

Reply:
La récupération des éléments permet de pouvoir gérer le bonne ordre d'affichage (label en premier, error en dernier, etc.). J'avoue me poser également la question avec la liberté que peut offrir le `children`. Je pense que c'est un bon sujet à aborder à notre point Kitten sur les composants. Limitation du composant pour éviter les dérives de composition VS Liberté de composition avec potentiellement des problèmes de styles (marges, etc.).

En état ce composant permet de créer la composition dans Kitten pour ne plus la faire sur les apps.